src/wolfio.c

```diff
@@ -916,7 +916,7 @@ int wolfIO_TcpConnect(SOCKET_T* sockfd, const char* ip, word16 port, int to_sec)
 #ifdef USE_WINDOWS_API
     if (*sockfd == SOCKET_INVALID)
 #else
-    if (*sockfd <= SOCKET_INVALID)
+    if ((*sockfd <= SOCKET_INVALID) || (*sockfd > FD_SETSIZE))
```

Comment:
I believe the FD_SETSIZE check should be in `wolfIO_Select`, not here.

It looks like Windows also has `FD_SETSIZE` (defaults to 64)`.  https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-select

Note: This code is only used by CRL and OCSP only. Optionally can be included at compile time with `HAVE_HTTP_CLIENT`.